File: CodeForces/1-100/96/A/96A.ts
// Problem 96A: Football
// https://codeforces.com/contest/96/problem/A
// Compiled with: npx 9.2.0, tsc Version 5.9.3
// Executed with: node v18.20.4
// Linux version: Linux kernel 6.1.0-10-amd64
// Written on: October 23rd, 2025
// Codeforces language used: Cannot be submitted

import * as readline from 'node:readline/promises';
import { stdin as input, stdout as output } from 'node:process';

async function main() {
    const rl = readline.createInterface({ input, output });

    const football: string = await rl.question('');
    rl.close();

    if (football.includes("0000000") || football.includes("1111111")) {
        console.log("YES");
    } else {
        console.log("NO");
    }
}

main();

File: CodeForces/1-100/96/A/96A.py
# Problem 96A: Football
# https://codeforces.com/contest/96/problem/A
# Executed with: Python 3.11.2
# Linux version: Linux kernel 6.1.0-10-amd64
# Submitted on: September 26th, 2024
# Codeforces language used: PyPy 3.10 (7.3.15, 64bit)

string = str(input())
if "0000000" in string or "1111111" in string:
    print("YES")
else:
    print("NO")
